rust/noosphere-core/src/api/client.rs

```diff
@@ -529,3 +567,56 @@ where
         Ok(push_response)
     }
 }
+
+#[cfg(feature = "test-gateway")]
+fn apply_test_header(headers: &mut HeaderMap, forced_host_header: &reqwest::header::HeaderValue) {
+    use reqwest::header::HOST;
+    _ = headers.remove(HOST);
+    headers.insert(HOST, forced_host_header.to_owned());
+}
+
+#[cfg(feature = "test-gateway")]
+fn create_test_header(api_base: &Url, identity: &str) -> Result<reqwest::header::HeaderValue> {
+    let mod_identity = identity
+        .strip_prefix("did:key:")
+        .ok_or_else(|| anyhow!("Could not format HOST header for test-gateway."))?;
```

Comment:
`Host` or `host`, not `HOST` https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Host